signac/__main__.py

```diff
@@ -672,6 +674,36 @@ def main_update_cache(args):
         _print_err("Updated cache (size={}).".format(n))
 
 
+def main_migrate(args):
+    from .contrib.migrations import apply_migrations
+    from filelock import FileLock
+    project = get_project(_ignore_schema_version=True)
+
+    schema_version = version.parse(SCHEMA_VERSION)
+    config_schema_version = version.parse(project.config['schema_version'])
+
+    if config_schema_version > schema_version:
+        _print_err(
+            "The schema version of the project ({}) is newer than the schema "
+            "version supported by signac version {}: {}. Try updatign signac.".format(
+                config_schema_version, __version__, schema_version))
+    elif config_schema_version == schema_version:
+        _print_err(
+            "The schema version of the project ({}) is up to date. "
+            "Nothing to do.".format(config_schema_version))
+    elif args.yes or query_yes_no(
+        "Do you want to migrate this project's schema version from '{}' to '{}'? "
+        "WARNING: THIS PROCESS IS IRREVERSIBLE!".format(
+            config_schema_version, schema_version), 'no'):
+        lock = FileLock(project.fn('.SIGNAC_PROJECT_MIGRATION_LOCK'))
+        with lock:
+            apply_migrations(project)
```

Comment:
Should this be wrapped in a `try:` and remove the lock file on `finally:`?

Reply:
Yes, I think so.